### PR TITLE
boleto: fix boleto to use document number

### PR DESCRIPTION
**Pull request by piiih**
At the present moment, the customer document number isn't required to
create boleto transactions. This pull request fixes this by requiring
and sending it to pagar.me API.

### CODE REVIEW

File: API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ public function confirm()
     {
         $this->load->model('checkout/order');
         $order = $this->model_checkout_order->getOrder($this->session->data['order_id']);
+        $this->log->write($order);
```
Comment (murilohns) [inline API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: This log is necessary?

File: API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php [hunk 2]
```diff
@@ -59,31 +60,38 @@ public function callback()
...
-                $document_number = $this->removeSeparadores($customer['cpf']);
-                $customer_name = $order_info['payment_firstname'] . " " . $order_info['payment_lastname'];
-            } else {
-                $document_number = $this->removeSeparadores($customer['cnpj']);
-                $customer_name = $customer['razao_social'];
-            }
-            $numero = $order_info['payment_numero'];
-            $complemento = $order_info['payment_company'];
+
+        $this->log->write($customer);
```
Comment (murilohns) [inline API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: This log is necessary?

File: API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php [hunk 3]
```diff
@@ -107,16 +115,21 @@ public function payment()
                 'id_pedido' => $order_info['order_id'],
                 'loja' => $this->config->get('config_title'),
             )));
+
+        $errorMessage = '';
```
Comment (murilohns) [inline API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: You need this line?

File: API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php [hunk 4]
```diff
@@ -107,16 +115,21 @@ public function payment()
...
+
         try {
             $transaction->charge();
         } catch (Exception $e) {
             $this->log->write("Erro Pagar.Me boleto: " . $e->getMessage());
-            die();
+            $errorMessage = $e->getMessage();
```
Comment (murilohns) [inline API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: I think you can use:
```
$this->log->write("Erro Pagar.Me boleto: " . $e->getMessage());
$json['error'] = $e->getMessage();
```

Comment (piiih) [reply API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: I think too, but I did this way to centralize the things related to errors inside of the `if/else` that sets `$json` variable. What do you think?

File: API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php (hunk 3 above)
Comment (piiih) [reply API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: PHP display warnings when you use a variable in some place and it wasn't declared =/

Comment (devdrops) [reply API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: @murilohns unfortunately yes. As @piiih says, we need to initialize this variable before the next steps, avoiding PHP to display a Notice message.

File: API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php (hunk 4 above)
Comment (murilohns) [reply API/1.5.x/upload/catalog/controller/payment/pagar_me_boleto.php]: the variable `$errorMessage` is just to save the `$e->getMessage();` and then set it to `json['error']`. 
Why not do it directly? `json['error'] = $e->getMessage();`

**Pull request merged**